test/fixtures/translator-test.js

```diff
@@ -81,6 +81,7 @@ export async function createTranslatorTest(htmlFilePath, dictionaryDirectory, di
     const test = createDomTest(htmlFilePath);
     const translator = await createTranslatorContext(dictionaryDirectory, dictionaryName);
     /** @type {import('vitest').TestAPI<{window: import('jsdom').DOMWindow, translator: Translator}>} */
+    // eslint-disable-next-line sonarjs/prefer-immediate-return
```

Comment:
There are a few cases where I had to disable `sonarjs/prefer-immediate-return` because it needs the type hint. I could have used a cast, but a type hint is a bit safer and the syntax a bit cleaner.